libraries/llvm/buffer.c

```diff
@@ -0,0 +1,141 @@
+#ifndef EFFEKT_BUFFER_C
+#define EFFEKT_BUFFER_C
+
+
+// TODO It may be performance-advantageous to implement this C file's semantics
+// in LLVM, since the linker cannot realistically be asked to satisfactorily
+// inline.
+
+
+// Eight bytes for the reference counter.
+// E.g. capacity could also be stored left of the data.
+#define BUFFER_METADATA_WIDTH (8)
```

Comment:
Call this `HEADER` instead of `METADATA` and make the pointer point to the reference count instead of the beginning of the bytes. Currently all our heap objects point to a reference count, and I prefer it if pointers point to something we've allocated.

Reply:
Done: [6c662db](https://github.com/effekt-lang/effekt/pull/142/commits/6c662db0d51a714fd4373c8e3c9778851ddf1304)